Sources/AppCoinsSDK/UI/Purchase/ViewModels/TransactionViewModel.swift

```diff
@@ -61,56 +62,63 @@ internal class TransactionViewModel : ObservableObject {
         bottomSheetViewModel.setPurchaseState(newState: .paying)
         
         if let product = product, let domain = domain {
-            walletUseCases.getWallet() {
-                result in
+            
+            self.getSupportedCurrencies(currency: product.priceCurrency) { 
```

Comment:
This should be done inside the getBalance method, and should not be done with getSupportedCurrencies but with getUserCurrency